File: python/链表/1669. 合并两个链表.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        pre,cur=None,list1
        i = 1
        while cur:
            if i == a:
                while cur and i <= b:
                    cur = cur.next
                    i+=1
                break
            if pre == None:
                pre = cur
            else:
                pre = pre.next
            cur = cur.next
            i+=1
        pre.next = list2
        while list2.next:
            list2 = list2.next
        list2.next = cur
        return list1

File: go/link/1669. 合并两个链表.go
package link

func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
	var pre *ListNode
	cur := list1
	i := 0
	for cur != nil {
		if i == a {
			for cur != nil && i <= b {
				cur = cur.Next
				i += 1
			}
			break
		}
		if pre == nil {
			pre = cur
		} else {
			pre = pre.Next
		}
		cur = cur.Next
		i += 1
	}
	pre.Next = list2
	for list2.Next != nil {
		list2 = list2.Next
	}
	list2.Next = cur
	return list1
}
